packages/react/src/components/List/ListItem.tsx

```diff
@@ -63,130 +77,185 @@ export interface ListItemProps
   onClick?: ComponentEventHandler<ListItemProps>
 }
 
-class ListItem extends UIComponent<WithAsProp<ListItemProps>> {
-  static create: ShorthandFactory<ListItemProps>
+const ListItem: React.FC<WithAsProp<ListItemProps> & { index: number }> &
+  FluentComponentStaticProps<ListItemProps> & {
+    slotClassNames: ListItemSlotClassNames
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext)
+  const { setStart, setEnd } = useTelemetry(ListItem.displayName, context.telemetry)
 
-  static displayName = 'ListItem'
+  setStart()
 
-  static className = 'ui-list__item'
+  const {
+    accessibility,
+    className,
+    content,
+    contentMedia,
+    design,
+    endMedia,
+    header,
+    important,
+    headerMedia,
+    media,
+    styles,
+    debug,
+    navigable,
+    selectable,
+    selected,
+    truncateContent,
+    truncateHeader,
+    variables,
+  } = props
 
-  static slotClassNames: ListItemSlotClassNames
+  const getA11Props = useAccessibility(accessibility, {
+    debugName: ListItem.displayName,
+    actionHandlers: {
+      performClick: e => {
+        e.preventDefault()
+        handleClick(e)
+      },
+    },
+    mapPropsToBehavior: () => ({
+      navigable,
+      selectable,
+      selected,
+    }),
+    rtl: context.rtl,
+  })
+  const { classes, styles: resolvedStyles } = useStyles(ListItem.displayName, {
+    className: ListItem.className,
+    mapPropsToStyles: () => ({
+      debug,
+      navigable,
+      important,
+      selectable,
+      selected,
+      truncateContent,
+      truncateHeader,
 
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      content: false,
+      hasContent: !!content,
+      hasContentMedia: !!contentMedia,
+      hasHeader: !!header,
+      hasHeaderMedia: !!headerMedia,
     }),
-    contentMedia: PropTypes.any,
-    content: PropTypes.any,
+    mapPropsToInlineStyles: () => ({ className, design, styles, variables }),
+    rtl: context.rtl,
+  })
 
-    debug: PropTypes.bool,
+  const ElementType = getElementType(props)
+  const unhandledProps = getUnhandledProps(ListItem.handledProps, props)
 
-    header: PropTypes.any,
-    endMedia: PropTypes.any,
-    headerMedia: PropTypes.any,
+  const handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
+    _.invoke(props, 'onClick', e, props)
+  }
 
-    important: PropTypes.bool,
-    media: PropTypes.any,
+  const contentElement = Box.create(content, {
```

Comment:
It's odd that we are not propagating any accessibility props to any of the slots :\

Reply:
I propose to fix this separately, may be in `implementsShorthand` test

Reply:
As it was like that before, I agree